### PR TITLE
Move telemetry client classes into separate module

### PR DESCRIPTION
**Pull request by sophokles73**
This addresses #2540

The current adapter client module is quite opinionated about its usage
scenario, i.e. it is intended to address requirements of protocol
adapters. However, other services also need to publish events but do not
need to query the registry not do they need to consume commands.

In order to better reflect such cases, the classes for sending Telemetry
and Event messages have been moved into dedicated telemetry,
telemetry-amqp and telemetry-kafka modules.

Subsequent PRs can be used to factor out the registry and C&C clients into
dedicated modules if desirable.

### CODE REVIEW

File: clients/telemetry/src/main/java/org/eclipse/hono/client/telemetry/EventSender.java [hunk 1]
```diff
@@ -12,7 +12,7 @@
  */
 
 
-package org.eclipse.hono.adapter.client.telemetry;
+package org.eclipse.hono.client.telemetry;
```
Comment (kaniyan) [inline clients/telemetry/src/main/java/org/eclipse/hono/client/telemetry/EventSender.java:15]: How about updating copyright notice with current year?

File: clients/telemetry/src/main/java/org/eclipse/hono/client/telemetry/TelemetrySender.java [hunk 2]
```diff
@@ -12,7 +12,7 @@
  */
 
 
-package org.eclipse.hono.adapter.client.telemetry;
+package org.eclipse.hono.client.telemetry;
```
Comment (kaniyan) [inline clients/telemetry/src/main/java/org/eclipse/hono/client/telemetry/TelemetrySender.java:15]: How about updating copyright notice with current year?

File: clients/telemetry/src/main/java/org/eclipse/hono/client/telemetry/EventSender.java (hunk 1 above)
Comment (sophokles73) [reply clients/telemetry/src/main/java/org/eclipse/hono/client/telemetry/EventSender.java:15]: well, I haven't changed the classes but merely renamed the package. But I am glad to adapt the year as well ...

File: clients/telemetry/src/main/java/org/eclipse/hono/client/telemetry/TelemetrySender.java (hunk 2 above)
Comment (kaniyan) [reply clients/telemetry/src/main/java/org/eclipse/hono/client/telemetry/TelemetrySender.java:15]: The same is also applicable to `ConnectionEventProducer` and `HonoEventConnectionEventProducerTest`.

Comment (sophokles73) [reply clients/telemetry/src/main/java/org/eclipse/hono/client/telemetry/TelemetrySender.java:15]: I have adapted the headers accordingly ...

**Review by kaniyan — APPROVED** [review]
LGTM

**Pull request merged**